### ISSUE TITLE
[Markdown] Table within list doesn't render correctly

### ISSUE THREAD

**Issue by ajeetdsouza** [rust-lang/crates.io#3501]
In the README of my crate, a Markdown table placed under a list element doesn't render correctly, even with the correct indentation. This works fine on GitHub, and it would be nice if crates.io was able to support this too.

For context, look at the `Environment Variables` section in both of these:
- GitHub README: https://github.com/ajeetdsouza/zoxide#environment-variables
- crates.io README: https://crates.io/crates/zoxide

---

**The following is an example of a table within a list in Markdown:**

```markdown
- Root 1
  - Item 1
  - Item 2:
    | OS          | Path                                     | Example                                    |
    | ----------- | ---------------------------------------- | ------------------------------------------ |
    | Linux / BSD | `$XDG_DATA_HOME` or `$HOME/.local/share` | `/home/alice/.local/share`                 |
    | macOS       | `$HOME/Library/Application Support`      | `/Users/Alice/Library/Application Support` |
    | Windows     | `{FOLDERID_RoamingAppData}`              | `C:\Users\Alice\AppData\Roaming`           |
```

**It should render this way:**

- Root 1
  - Item 1
  - Item 2:
    | OS          | Path                                     | Example                                    |
    | ----------- | ---------------------------------------- | ------------------------------------------ |
    | Linux / BSD | `$XDG_DATA_HOME` or `$HOME/.local/share` | `/home/alice/.local/share`                 |
    | macOS       | `$HOME/Library/Application Support`      | `/Users/Alice/Library/Application Support` |
    | Windows     | `{FOLDERID_RoamingAppData}`              | `C:\Users\Alice\AppData\Roaming`           |

**Comment by plaets** [comment]
Looks like this could be a bug in [comrak](https://crates.io/crates/comrak), the markdown library used by crates.io. 

You can test this by installing comrak using `cargo install comrak` and then running something like 
`cat README.md | comrak -e autolink -e strikethrough -e table -e tagfilter -e tasklist --unsafe --header-ids user-content-` 
to manually render your readme to html (options from [src/render.rs](https://github.com/rust-lang/crates.io/blob/df443dbcdd6ad2e1cb3e422197e823009c4aba47/src/render.rs)). The relevant part of the result should look like this
![image](https://user-images.githubusercontent.com/49844988/114093745-c7017580-98bb-11eb-92c7-c97504bcdd25.png)
[Full output here](https://gist.github.com/plaets/d5301ed75a48b2db81572805bfcee6e9)

**Comment by ajeetdsouza** [comment]
@plaets thanks for the quick reply, I've created an issue here: https://github.com/kivikakk/comrak/issues/182.

**Comment by kivikakk** [comment]
Hi! I've released [0.10.1-rc.1](https://crates.io/crates/comrak/0.10.1-rc.1) which fixes this issue. Sorry for the rc but I just don't have the time for a full release right now. This specifically fixes this issue, per the same change made in upstream cmark-gfm.
